client/state/ui/notices/reducer.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import { filter } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	NEW_NOTICE,
+	REMOVE_NOTICE
+} from 'state/action-types';
+
+/**
+ * Tracks all known site objects, indexed by site ID.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function items( state = [], action ) {
+	switch ( action.type ) {
+		case NEW_NOTICE:
+			state = [ action, ...state ];
+			break;
+		case REMOVE_NOTICE:
+			state = filter( state, ( notice ) => ( notice.noticeId !== action.noticeId ) );
```

Comment:
Also, while Lodash's functions have sometimes been shown to be faster than their native equivalents, you don't need a module for `filter`:

```jsx
state = state.filter( ( notice ) => ( notice.noticeId !== action.noticeId ) );
```